testsuite/internal.ml

```diff
@@ -1,5 +1,65 @@
 #use "internal_test_header.ml";;
 
+
+let () = test "Preprocess"
+    ~description:"Check that preprocessor works"
+    ~requirements:ocamlopt_available
+    ~options:[]
+    ~tree:[
+      T.f "main.ml" ~content:{|let () = Printf.printf "line %d\n" __LINE__|};
+      T.f "preprocessor.ml" ~content:{|
+let () =
+  let all = Array.to_list Sys.argv |> List.tl in
+  let input = Sys.argv.(3) in
+  let txt = Sys.argv.(1) in
+  let shift = int_of_string (Sys.argv.(2)) in
+  let ic = open_in input in
+  for i = 1 to shift do
+    print_endline (Printf.sprintf "(* shift lines by one, %s *)" txt);
+  done;
+  (try
+  while true do
+    let l = input_line ic in
+    print_endline l;
+  done;
+  with End_of_file -> ());
+  print_endline {txt|let () = if __LINE__ <> 4 then failwith "unexpected shift" |txt};
+  print_endline (Printf.sprintf {txt|
+let () =
+  match [%s] with
+  | ["with space"; "2"; _file] -> ()
+  | l -> failwith (Printf.sprintf "Preprocessor received unexpected args: %%s" (String.concat " - " l))
+|txt} (String.concat "; " (List.map (Printf.sprintf "%S") all)))
+
+|};
+      T.f "myocamlbuild.ml" ~content:{|
+open Ocamlbuild_plugin
+let () =
+  dispatch begin function
+  | After_rules ->
+    dep ["mypreprocessor"] ["preprocessor.exe"];
+    flag ["ocaml"; "pp"; "mypreprocessor"] (S [P "./preprocessor.exe";
+                                               A "with space";
+                                               A "2"]);
+    rule "native to exe"
+    ~prods:["%.exe"]
+    ~dep:"%.native"
+    (fun env _build ->
+      let nat = env "%.native" and exe = env "%.exe" in
+      Cmd(S[A "cp"; A nat; A exe]))
+  | _ -> ()
+  end
+|};
+      T.f "_tags" ~content:{|
+<main.ml>: mypreprocessor
+|}]
+    ~targets:("main.native", [])
+    ~post_cmd:(
+      if Sys.win32
+      then "bash --norc -c ./main.native"
```

Comment:
It is odd that we have to do this here. Should the function that runs `post_cmd` be calling `My_std.sys_command` instead?

Reply:
The testsuite doesn't currently have access to ocamlbuild the lib. Should I make the change ?

Reply:
I'm okay with keeping this file separate from the ocamlbuild logic, as an external driver, but I would rather have the Windows-testing logic in `ocamlbuild_test.ml` rather than in each caller of the `test` function.

Reply:
done